src/psyclone/transformations.py

```diff
@@ -488,7 +491,7 @@ def validate_it_can_run_on_gpu(self, node, options):
 
         calls = kernel_schedule.walk(Call)
         for call in calls:
-            if not call.is_available_on_device():
+            if not call.is_available_on_device(device_string):
                 call_str = call.debug_string().rstrip("\n")
                 raise TransformationError(
                     f"{k_or_r} '{node.name}' calls another routine "
```

Comment:
Please extend the error message both here and at L1776 to mention the value of 'device_str' if it is set.